be/src/util/compression/block_compression.cpp

```diff
@@ -966,6 +967,34 @@ class GzipBlockCompression final : public ZlibBlockCompression {
     const static int MEM_LEVEL = 8;
 };
 
+class GzipBlockCompressionV2 final : public GzipBlockCompression {
+public:
+    GzipBlockCompressionV2() : GzipBlockCompression() {}
+
+    static const GzipBlockCompressionV2* instance() {
+        static GzipBlockCompressionV2 s_instance;
+        return &s_instance;
+    }
+
+    ~GzipBlockCompressionV2() override = default;
+
+    Status decompress(const Slice& input, Slice* output) const override {
+        thread_local libdeflate_decompressor* decompressor = libdeflate_alloc_decompressor();
```

Comment:
why `thread_local`?

Reply:
because `libdeflate_decompressor` is not safe to be use by different thread concurrently and I wanna avoid allocate them each time I invoke this method, so `thread_local` help me do that because it imply `static` semantic

Reply:
suggest to move this thread_local to the field of the class